### ISSUE TITLE
Celluloid in Linux Mint 22.0 (Wayland) 

### ISSUE THREAD

**Issue by callagg2** [celluloid-player/celluloid#1000]
Overview Description:
Any mp4 video appears with a red filter overlaying it when played in Celluloid in Wayland (but fine in MPV & VLC). Also, appears fine in Celluloid in X11 .

Steps to Reproduce:
1. Open the Wayland version of Linux Mint Cinnamon 22.0 (and 22.1 beta)
2. In Terminal, open an mp4 file using Celluloid

Actual Results:
Video is corrupted with a red filter

Expected Results:
Video should appear with no red filter overlaying it

Version: apt policy celluloid
celluloid:
  Installed: 0.21+mint2+wilma
  Candidate: 0.21+mint2+wilma
  Version table:
     0.26-1build2 500
        500 https://mirror.webworld.ie/ubuntu noble/universe amd64 Packages
 *** 0.21+mint2+wilma 700
        700 https://mirror.webworld.ie/linuxmint/packages xia/upstream amd64 Packages
        100 /var/lib/dpkg/status

Additional Information:
In the X11 version of Linux Mint 22.0 - everything is fine
<script src="https://pastebin.com/embed_js/6niyUGbE"></script>

In the Wayland version of Linux Mint 22.0 - the red filter appears
<script src="https://pastebin.com/embed_js/6niyUGbE"></script>

**Comment by leandrocunha526** [comment]
Please, try to reproduce on the latest version possible with both 0.26 (Ubuntu 24.04 repository) and 0.27 (Flatpak).

**Comment by callagg2** [comment]
Leandro,
I can confirm that the flatpak version (0.27) works perfectly.
So, it's just the version in Linux Mint.

**Comment by leandrocunha526** [comment]
> Leandro, I can confirm that the flatpak version (0.27) works perfectly. So, it's just the version in Linux Mint.

You can pull the latest version by running `apt install celluloid=0.26-1build2` in this case to get the Ubuntu version (I worked on it on Debian).

celluloid:
  Instalado: 0.26-1build2
  Candidato: 0.26-1build2
  Tabela de versão:
 *** 0.26-1build2 500
        500 http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages
        100 /var/lib/dpkg/status
     0.21+mint2+wilma 700
        700 http://packages.linuxmint.com xia/upstream amd64 Packages

**Comment by callagg2** [comment]
Yeah, that installed fine, Celluloid is now version 0.26 .
No errors messages installing it

The following packages will be upgraded:
  celluloid
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 325 kB of archives.
After this operation, 540 kB of additional disk space will be used.
Get:1 https://mirror.webworld.ie/ubuntu noble/universe amd64 celluloid amd64 0.26-1build2 [325 kB]
Fetched 325 kB in 0s (820 kB/s)   
(Reading database ... 466922 files and directories currently installed.)
Preparing to unpack .../celluloid_0.26-1build2_amd64.deb ...
Unpacking celluloid (0.26-1build2) over (0.21+mint2+wilma) ...
Setting up celluloid (0.26-1build2) ...
Processing triggers for libglib2.0-0t64:amd64 (2.80.0-6ubuntu3.2) ...
Processing triggers for mailcap (3.70+nmu1ubuntu1) ...
Processing triggers for desktop-file-utils (0.27-2build1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for gnome-menus (3.36.0-1.1ubuntu3) ...
Processing triggers for mate-menus (1.26.1+mint1) ...
Processing triggers for man-db (2.12.0-4build2) ..

It also open fine, looks a lot more modern than the previous version.
but the video still has a red layer overlaying it :-(
Launching Celluloid in Terminal, I get two messages repeated several times
"Gsk-Message: 22:10:06.576: Failed to realize renderer of type 'GskNglRenderer' for surface 'GdkWaylandToplevel': OpenGL ES 3.0 is not supported by this renderer."
"Cannot load libcuda.so.1@
"Gsk-Message: 22:10:12.454: Failed to realize renderer of type 'GskNglRenderer' for surface 'GdkWaylandPopup': OpenGL ES 3.0 is not supported by this renderer."

Any advice?

**Comment by leandrocunha526** [comment]
> Yeah, that installed fine, Celluloid is now version 0.26 . No errors messages installing it
> 
> The following packages will be upgraded: celluloid 1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded. Need to get 325 kB of archives. After this operation, 540 kB of additional disk space will be used. Get:1 https://mirror.webworld.ie/ubuntu noble/universe amd64 celluloid amd64 0.26-1build2 [325 kB] Fetched 325 kB in 0s (820 kB/s) (Reading database ... 466922 files and directories currently installed.) Preparing to unpack .../celluloid_0.26-1build2_amd64.deb ... Unpacking celluloid (0.26-1build2) over (0.21+mint2+wilma) ... Setting up celluloid (0.26-1build2) ... Processing triggers for libglib2.0-0t64:amd64 (2.80.0-6ubuntu3.2) ... Processing triggers for mailcap (3.70+nmu1ubuntu1) ... Processing triggers for desktop-file-utils (0.27-2build1) ... Processing triggers for hicolor-icon-theme (0.17-2) ... Processing triggers for gnome-menus (3.36.0-1.1ubuntu3) ... Processing triggers for mate-menus (1.26.1+mint1) ... Processing triggers for man-db (2.12.0-4build2) ..
> 
> It also open fine, looks a lot more modern than the previous version. but the video still has a red layer overlaying it :-( Launching Celluloid in Terminal, I get two messages repeated several times "Gsk-Message: 22:10:06.576: Failed to realize renderer of type 'GskNglRenderer' for surface 'GdkWaylandToplevel': OpenGL ES 3.0 is not supported by this renderer." "Cannot load libcuda.so.1@ "Gsk-Message: 22:10:12.454: Failed to realize renderer of type 'GskNglRenderer' for surface 'GdkWaylandPopup': OpenGL ES 3.0 is not supported by this renderer."
> 
> Any advice?

Mint, Ubuntu and Debian's GTK4 sets GSK_RENDERER to ngl and you can set it to gl and see if the problem persists.

- https://discussion.fedoraproject.org/t/gsk-renderer-ngl-and-gl-nvidia-xorg-random-gsk-errors-after-6-7-apps-open/135123
- https://bbs.archlinux.org/viewtopic.php?id=294177&p=2
- https://www.phoronix.com/news/GTK-4.16-Released
- https://bugs.launchpad.net/ubuntu/+source/gtk4/+bug/2061079
- https://launchpad.net/bugs/2082017

**Comment by callagg2** [comment]
Yeah, unfortunately i even tried adding GDK_DEBUG=gl-prefer-gl and GSK_RENDERER=gl  to /etc/environment and rebooting, but no luck. 
ah, it's ok, I'll default back to X11 on Mint, at least I know that definitely works.

Thanks for trying though, while I may not have solved it, 
at least i know it's my GPU at fault, and the flatpak version works :-)

**Comment by leandrocunha526** [comment]
> Yeah, unfortunately i even tried adding GDK_DEBUG=gl-prefer-gl and GSK_RENDERER=gl to /etc/environment and rebooting, but no luck. ah, it's ok, I'll default back to X11 on Mint, at least I know that definitely works.
> 
> Thanks for trying though, while I may not have solved it, at least i know it's my GPU at fault, and the flatpak version works :-)

I use Debian testing, Celluloid doesn't crash for me with ngl and errands already crashes if I don't put ngl under Xorg.
I only tested it in X11 environments too, both XFCE and Mate.
